app/models/concerns/publishable.rb

```diff
@@ -33,13 +33,16 @@ def published?
     published_at && published_at == edited_at
   end
 
-  def purge!
-    production_fedora_connection.purge_object(pid: pid) rescue RestClient::ResourceNotFound
-    update_attributes(state: STATE_DELETED) # This is a soft-delete
-  end
+  def purge!(user_id = nil)
+    user = User.find(user_id) if user_id
 
-  def purged?
-    state == STATE_DELETED
+    if destroy_published_version!
```

Comment:
Will this (along with ```destroy_draft_version!```) be enough to remove everything? Or should I be using ```purge_object``` with the published and draft pids?